binary_search_tree/tree.py

```diff
@@ -14,44 +14,160 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)
+    def add(self, key, value):
+        new_node = TreeNode(key,value)
+        if not self.root:
+            self.root = new_node
+            return 
+        current = self.root
+        previous = None
+        while current:
+            previous = current
+            if current.key > new_node.key:
+                current = current.left
+            else:
+                current = current.right
+        if previous.key > new_node.key:
+            previous.left = new_node
+        else:
+            previous.right = new_node
```

Comment:
You are storing things that are less than the current node on the **right** side and elements greater than on the left side.  This works, but you end up with an inverted tree.

That's why your postorder isn't working properly on the tests.